src/main/kotlin/ua/marchenko/artauction/common/reactive/ReactiveExtension.kt

```diff
@@ -0,0 +1,5 @@
+package ua.marchenko.artauction.common.reactive
+
+import reactor.core.publisher.Mono
+
+fun <T> Mono<T>.switchIfEmpty(s: () -> Mono<T>): Mono<T> = this.switchIfEmpty(Mono.defer { s() })
```

Comment:
why do we custom implementation?

You can use `reactor.kotlin.core.publisher.switchIfEmpty` (from kotlin extension lib that already applied to your project)